File: packages/tf-lang-l0-ts/src/ops/lens_mod.ts
export function lens_mod(x: unknown, m: unknown): number | null {
  if (typeof x !== 'number' || typeof m !== 'number' || !Number.isInteger(x) || !Number.isInteger(m)) {
    return null;
  }
  if (m <= 0) return null;
  const r = ((x % m) + m) % m;
  return r;
}

File: packages/tf-lang-l0-rs/src/ops/lens_mod.rs
use anyhow::{bail, Result};
use serde_json::Value;

pub fn lens_mod(x: &Value, m: &Value) -> Result<Value> {
    let xi = x.as_i64().ok_or_else(|| anyhow::anyhow!("E_MOD_TYPE"))?;
    let mi = m.as_i64().ok_or_else(|| anyhow::anyhow!("E_MOD_TYPE"))?;
    if mi <= 0 {
        bail!("E_MOD_BOUNDS");
    }
    let r = ((xi % mi) + mi) % mi;
    Ok(Value::Number(r.into()))
}
